### DISCUSSION TITLE
How to include an additional element within a loop without adding an additional level?

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3802] [1. Help]
I have the following loop `<template x-for="(w, index) in webcams">`. Now in case index reaches a certain number, I need to include an additional html element. However, how can I do that, since I can only have one html element inside a `<template>` tag. And I can't have a surrounding html tag because the layout uses grid and the additional element needs to be spanned over all columns in that row.

How can I achieve this without the surrounding `<span>` tag?

![image](https://github.com/alpinejs/alpine/assets/1435963/f623264f-c32d-4f40-9a47-6d554ec9b477)

**Comment by ekwoka** [comment]
you can have the first child of the template be a div/span with [`display: contents`](https://developer.mozilla.org/en-US/docs/Web/CSS/display#box)

This basically makes the element a passthrough, not unlike a `fragment` in JSX.

🤔 Maybe it would be worth auto wrapping elements. I'd have to check if there are strange CSS rules that don't like it...

But when yuou can control the css, that works fine. I've never actually run into a case where css had issues. It basically jsut makes that element not exist as far as layout and styling are concerned

**Reply by simonferndriger** [reply]
Works like a charm! 🥳 Yes, I actually would encourage to make this a default case. That way, multiple child elements could be rendered within a template 😊
